test/verify/check-s4u-ssh

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/python3
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2020 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import parent
+
+from testlib import *
+
+@skipImage("No realmd available", "fedora-coreos")
+@skipImage("freeipa not currently in testing", "debian-testing")
+class TestS4USsh(MachineCase):
+    provision = {
+        "0": {"address": "10.111.113.1/20", "dns": "10.111.112.100"},
+        "1": {"address": "10.111.113.2/20", "dns": "10.111.112.100"},
+        "services": {"image": "services", "memory_mb": 2048}
+    }
+
+    def setUp(self):
+        super().setUp()
+        self.machines['services'].execute("/run-freeipa")
+
+    def testBasic(self):
+        client_machine = self.machine
+        sshd_machine = self.machines["1"]
+        ipa_machine = self.machines["services"]
+
+        # set hostname for readable logs
+        client_machine.execute("hostnamectl set-hostname sshclient.cockpit.lan")
+        sshd_machine.execute("hostnamectl set-hostname sshserver.cockpit.lan")
+        self.login_and_go("/system")
+
+        # join both machines
+        wait(lambda: client_machine.execute("nslookup -type=SRV _ldap._tcp.cockpit.lan"))
+        wait(lambda: sshd_machine.execute("nslookup -type=SRV _ldap._tcp.cockpit.lan"))
+        client_machine.execute("echo foobarfoo | realm join -U admin cockpit.lan")
+        sshd_machine.execute("echo foobarfoo | realm join -U admin cockpit.lan")
+
+        # configure ipa
+        ipa_machine.execute("docker exec freeipa bash -c 'echo foobarfoo | kinit admin@COCKPIT.LAN'")
+        # add user to impersonate
+        ipa_machine.execute("docker exec freeipa bash -c 'ipa user-add --first=user --last=user user'")
+        # add service which will be allowed to delegate creds
+        ipa_machine.execute("docker exec freeipa bash -c 'ipa service-add cockpitclient/sshclient.cockpit.lan@COCKPIT.LAN --ok-as-delegate=true --ok-to-auth-as-delegate=true'")
+        # Allow restrieval of service/host keytabs by admin
+        ipa_machine.execute("docker exec freeipa bash -c 'ipa service-allow-retrieve-keytab --user=admin cockpitclient/sshclient.cockpit.lan'")
+        # set up delegation rule
+        ipa_machine.execute(script="""#!/bin/sh
```

Comment:
I wasn't even aware of `script=` :grin: 

I actually meant something like
```
script = """ipa service ....
ipa ...
"""
ipa_machine.execute("docker exec freeipa bash -ec '%s'" % script)
```

Like this it also avoids most of the overhead, but at least needs a `set -e`.

Reply:
oh hahaha, i'll do that.

And ye I have the other fixes locally.